openwisp_monitoring/device/api/views.py

```diff
@@ -423,6 +444,75 @@ def _create_access_tech_chart(self, metric):
 device_metric = DeviceMetricView.as_view()
 
 
+class WifiSessionFilter(filters.FilterSet):
+    organization_slug = filters.CharFilter(
+        field_name='device__organization__slug', label='Organization slug'
+    )
+
+    class Meta:
+        model = WifiSession
+        fields = {
+            'device': ['exact'],
+            'device__group': ['exact'],
+            'start_time': ['exact', 'gt', 'gte', 'lt', 'lte'],
+            'stop_time': ['exact', 'gt', 'gte', 'lt', 'lte'],
+        }
+
+
+class WifiSessionListCreateView(ProtectedAPIMixin, ListCreateAPIView):
+    queryset = WifiSession.objects.select_related(
+        'device', 'wifi_client', 'device__organization', 'device__group'
+    )
+    organization_field = 'device__organization'
+    filter_backends = [DjangoFilterBackend]
+    pagination_class = ListViewPagination
+    filterset_class = WifiSessionFilter
+
+    def get_serializer_class(self):
+        if self.request.method in ['GET']:
+            return WifiSessionReadSerializer
+        return WifiSessionCreateUpdateSerializer
+
+
+wifi_session_list = WifiSessionListCreateView.as_view()
+
+
+class WifiSessionDetailView(ProtectedAPIMixin, RetrieveUpdateAPIView):
```

Comment:
Why aren't we allowing to delete objects here?

In general through the API we allow a bit more flexibility to allow to extend the capabilities of OpenWISP if any user needs it.